python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -224,23 +233,32 @@ def get_asset_run_submit_input(
         )
     )
 
-    external_job = code_location.get_external_job(
-        JobSubsetSelector(
-            location_name=location_name,
-            repository_name=repository_name,
-            job_name=job_name,
-            op_selection=None,
-            asset_selection=asset_keys,
-        )
+    job_selector = JobSubsetSelector(
+        location_name=location_name,
+        repository_name=repository_name,
+        job_name=job_name,
+        op_selection=None,
+        asset_selection=asset_keys,
     )
 
-    external_execution_plan = code_location.get_external_execution_plan(
-        external_job,
-        run_request.run_config,
-        step_keys_to_execute=None,
-        known_state=None,
-        instance=instance,
-    )
+    selector_id = hash_collection(job_selector)
+
+    if selector_id not in pipeline_and_execution_plan_cache:
+        external_job = code_location.get_external_job(job_selector)
+
+        external_execution_plan = code_location.get_external_execution_plan(
+            external_job,
+            {},
```

Comment:
tiny nit: just passing this as `run_config={}` makes the code a bit clearer (to my untrained eye)